pkg/tenant/db/errors.go

```diff
@@ -49,22 +54,44 @@ var (
 	ErrTopicNameTooLong          = errors.New("topic name is too long")
 )
 
+// ValidationError represents a validation error for a specific field and may contain
+// an index if the field is a slice.
 type ValidationError struct {
-	err error
+	Field string
+	Err   error
+	Index int
```

Comment:
This data structure will have to be JSON serialized for the front end:

```suggestion
	Field string `json:"field"`
	Err   error `json:"error"
	Index int `json:"index"
```

Reply:
I was thinking of adding a similar struct to the API package, except that it contains the JSON tags for serialization, and then converting the one here into the API response struct. But if that's too clunky we could just add the tags here.

Reply:
That makes sense to me - whatever you'd prefer; just making sure I wasn't missing something in translation!